packaging/tags.py

```diff
@@ -26,14 +24,39 @@
 _32_BIT_INTERPRETER = sys.maxsize <= 2 ** 32
 
 
-@attr.s(frozen=True, repr=False)
 class Tag(object):
-    interpreter = attr.ib(converter=str.lower)
-    abi = attr.ib(converter=str.lower)
-    platform = attr.ib(converter=str.lower)
+
+    __slots__ = ["_interpreter", "_abi", "_platform"]
+
+    def __init__(self, interpreter, abi, platform):
+        self._interpreter = str.lower(interpreter)
+        self._abi = str.lower(abi)
+        self._platform = str.lower(platform)
+
+    @property
```

Comment:
Is there a need for `property` and to have hidden attributes instead of simply directly exposing e.g. `interpreter`?

Reply:
I've used `property` to mimic the previous behavior as closely as possible.

Reply:
My understanding is that it is for making the public attributes read-only, which is a good idea IMO.

Reply:
@pradyunsg in general I would argue otherwise, but since this implements `__hash__` the class the should do a reasonable job of trying  to be immutable without trying to work too hard. So once again, @pradyunsg wins. :)